modules/layers/src/icon-layer/icon-layer.js

```diff
@@ -194,6 +174,73 @@ export default class IconLayer extends Layer {
     );
   }
 
+  _updateIconMapping(iconMapping) {
+    const attributeManager = this.getAttributeManager();
+    attributeManager.invalidate('instanceOffsets');
+    attributeManager.invalidate('instanceIconFrames');
+    attributeManager.invalidate('instanceColorModes');
+    this.setState({iconMapping});
+  }
+
+  _updatePrePacked({oldProps, props}) {
+    const {iconAtlas, iconMapping} = props;
+    const {gl} = this.context;
+    if (iconMapping && oldProps.iconMapping !== iconMapping) {
+      this._updateIconMapping(iconMapping);
+    }
+
+    if (oldProps.iconAtlas !== iconAtlas) {
+      if (iconAtlas instanceof Texture2D) {
+        iconAtlas.setParameters({
+          [GL.TEXTURE_MIN_FILTER]: DEFAULT_TEXTURE_MIN_FILTER,
+          [GL.TEXTURE_MAG_FILTER]: DEFAULT_TEXTURE_MAG_FILTER
+        });
+        this.setState({iconsTexture: iconAtlas});
+      } else if (typeof iconAtlas === 'string') {
+        loadTextures(gl, {
+          urls: [iconAtlas]
+        }).then(([texture]) => {
+          texture.setParameters({
+            [GL.TEXTURE_MIN_FILTER]: DEFAULT_TEXTURE_MIN_FILTER,
+            [GL.TEXTURE_MAG_FILTER]: DEFAULT_TEXTURE_MAG_FILTER
+          });
+          this.setState({iconsTexture: texture});
+        });
+      }
+    }
+  }
+
+  _updateAutoPacking({props}) {
+    const {data, getIcon, updateTriggers} = props;
+    const {gl} = this.context;
+    const attributeManager = this.getAttributeManager();
+
+    let iconManager = this.state.iconManager;
+    if (!iconManager) {
+      iconManager = new IconManager(gl, {
+        data,
+        getIcon,
+        onTextureUpdate: texture => this._onTextureUpdate(texture)
+      });
+      this.setState({
+        iconManager,
+        iconsTexture: iconManager.texture
+      });
+    }
+
+    iconManager.setData(props.data);
+    if (this.state.iconMapping !== iconManager.mapping) {
+      if (updateTriggers && updateTriggers.getIcon) {
```

Comment:
This is not how updateTriggers work. See https://github.com/uber/deck.gl/blob/master/modules/layers/src/path-layer/path-layer.js#L125